roles/resource_manager/includes/retrieve.yaml

```diff
@@ -1,32 +1,16 @@
 ---
-- name: Perform retrieve with provided credentials
-  when: data_store.scm.origin.token is defined
-  block:
-    - name: Retrieve the host vars with provided token access
-      ansible.scm.git_retrieve:
-        origin:
-          url: "{{ data_store['scm']['origin']['url'] }}"
-          token: "{{ data_store['scm']['origin']['token'] }}"
-        parent_directory: "{{ data_store.scm.parent_directory | default(resource_manager_inventory_directory) }}"
-      register: resource_manager_result
+- name: Retrieve the host vars with token access (if provided)
+  ansible.scm.git_retrieve:
+    origin:
+      url: "{{ data_store['scm']['origin']['url'] }}"
+      token: "{{ data_store['scm']['origin']['token'] | d(None) }}"
+    parent_directory: "{{ data_store.scm.parent_directory | default(resource_manager_inventory_directory) }}"
+  changed_when: false
```

Comment:
Added this to ensure that deploy operation reports changes ONLY when it actually pushes config lines. Right now, even if the actual deploy task is green, the retrieve operations returns changed: true. This might lead to unintended consequences when used in let's say, Controller Workflow or in any place where the outcome of this is used to trigger other jobs.